spark-3.1-spanner-lib/src/test/java/com/google/cloud/spark/spanner/SpannerSparkTest.java

```diff
@@ -0,0 +1,138 @@
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.DataTypes;
+import org.apache.spark.sql.types.StructField;
+import org.apache.spark.sql.types.StructType;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SpannerSparkTest {
+
+  String databaseId = "spark-db";
+  String instanceId = "spanner-spark";
+  String projectId = "orijtech-161805";
+  DatabaseAdminClient dbAdminClient;
+  Spanner spanner;
+  String emulatorHost = System.getenv("SPANNER_EMULATOR_HOST");
+
+  @Before
+  public void setUp() throws Exception {
+    String emulatorHost = System.getenv("SPANNER_EMULATOR_HOST");
+    SpannerOptions opts = SpannerOptions.newBuilder().setEmulatorHost(emulatorHost).build();
+    spanner = opts.getService();
+    // 1. Create the Spanner instance.
+    // TODO: Skip this process if the instance already exists.
+    InstanceAdminClient insAdminClient = spanner.getInstanceAdminClient();
+    InstanceConfig config = insAdminClient.listInstanceConfigs().iterateAll().iterator().next();
+    InstanceInfo insInfo =
+        InstanceInfo.newBuilder(InstanceId.of(projectId, instanceId))
+            .setInstanceConfigId(config.getId())
+            .setNodeCount(1)
+            .setDisplayName("SparkSpanner Test")
+            .build();
+    OperationFuture<Instance, CreateInstanceMetadata> iop = insAdminClient.createInstance(insInfo);
+
+    try {
+      iop.get();
+    } catch (Exception e) {
+      if (!e.toString().contains("ALREADY_EXISTS")) {
+        throw e;
+      }
+    }
+
+    dbAdminClient = spanner.getDatabaseAdminClient();
+    // 2. Create the database.
+    // TODO: Skip this process if the database already exists.
+    OperationFuture<Database, CreateDatabaseMetadata> dop =
+        dbAdminClient.createDatabase(
+            instanceId,
+            databaseId,
+            Arrays.asList(
+                "CREATE TABLE ATable (\n"
+                    + " A INT64 NOT NULL,\n"
+                    + " B STRING(100),\n"
+                    + " C BYTES(MAX),\n"
+                    + " D TIMESTAMP,\n"
+                    + " E NUMERIC,\n"
+                    + " F ARRAY<STRING(MAX)>\n"
+                    + ") PRIMARY KEY(A)"));
+    try {
+      dop.get();
+    } catch (Exception e) {
+      if (!e.toString().contains("ALREADY_EXISTS")) {
+        throw e;
+      }
+    }
+  }
+
+  @After
+  public void teardown() {
+    spanner.close();
+  }
+
+  private Map<String, String> connectionProperties() {
+    Map<String, String> props = new HashMap<>();
+    props.put("databaseId", databaseId);
+    props.put("instanceId", instanceId);
+    props.put("projectId", projectId);
+    if (false) {
+      props.put("emulatorHost", emulatorHost);
+    }
+    props.put("table", "ATable");
+    return props;
+  }
+
+  @Test
+  public void testSpannerTable() {
```

Comment:
I'd create a separate test file SpannerTableTest and  move the function testSpannerTable there.